File: kotlin/rna-transcription/src/test/kotlin/RnaTranscriptionTest.kt
import org.junit.Test
import org.junit.Ignore
import kotlin.test.assertEquals

class RnaTranscriptionTest {
    @Test
    fun `empty RNA` ()= assertEquals("", transcribeToRna(""))

    @Test
    fun `RNA complement of cytosine is guanine`() = assertEquals("G", transcribeToRna("C"))

    @Test
    fun `RNA complement of guanine is cytosine`() = assertEquals("C", transcribeToRna("G"))

    @Test
    fun `RNA complement of thymine is adenine`() = assertEquals("A", transcribeToRna("T"))

    @Test
    fun `RNA complement of adenine is uracil`() = assertEquals("U", transcribeToRna("A"))

    @Test
    fun `compound RNA`() = assertEquals("UGCACCAGAAUU", transcribeToRna("ACGTGGTCTTAA"))
}

File: scala/rna-transcription/src/test/scala/RnaTranscriptionTest.scala
import org.scalatest.{Matchers, FunSuite}

/** @version 1.2.0 */
class RnaTranscriptionTest extends FunSuite with Matchers {

  test("RNA complement of cytosine is guanine") {
    RnaTranscription.toRna("C") should be(Some("G"))
  }

  test("RNA complement of guanine is cytosine") {
    RnaTranscription.toRna("G") should be(Some("C"))
  }

  test("RNA complement of thymine is adenine") {
    RnaTranscription.toRna("T") should be(Some("A"))
  }

  test("RNA complement of adenine is uracil") {
    RnaTranscription.toRna("A") should be(Some("U"))
  }

  test("RNA complement") {
    RnaTranscription.toRna("ACGTGGTCTTAA") should be(Some("UGCACCAGAAUU"))
  }
}
